code/modules/hydroponics/eggincubator.dm

```diff
@@ -0,0 +1,108 @@
+/obj/machinery/egg_incubator
+	name = "egg incubator"
+	icon = 'icons/obj/virology.dmi'
+	icon_state = "incubator"
+	layer = 2.9
+	density = 1
+	anchored = 1
+	use_power = 1
+	idle_power_usage = 10
+	active_power_usage = 500
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE | EJECTNOTDEL
+	flags = OPENCONTAINER | NOREACT
+	pass_flags = PASSTABLE
+	var/list/egg_storage = list()
+	var/limit = 1
+	var/speed_bonus = 0
+
+/obj/machinery/egg_incubator/New()
+	. = ..()
+
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/egg_incubator,
+		/obj/item/weapon/stock_parts/capacitor,
+		/obj/item/weapon/stock_parts/capacitor,
+		/obj/item/weapon/stock_parts/matter_bin
+	)
+
+	RefreshParts()
+
+/obj/machinery/egg_incubator/RefreshParts()
+	var/bincount = 0
+	var/capcount = 0
+	for(var/obj/item/weapon/stock_parts/SP in component_parts)
+		if(istype(SP, /obj/item/weapon/stock_parts/capacitor)) capcount += SP.rating-1
+		if(istype(SP, /obj/item/weapon/stock_parts/matter_bin)) bincount += SP.rating

```

Comment:
It's on the base stock parts class probe.

Also, use `get_rating()` instead of the `rating` var